### PR TITLE
update (source): Establish chapter 3 dk script

### PR DESCRIPTION
**Pull request by acidmanifesto**
Per source notes:
```
Xinef: Should be in chapter III
```
We establish chapter 3 script for scarlet enclaved and to make the easy of future development in terms of organization
we are going to need it.

TO PEOPLE TESTING THIS PR PLEASE RE-RUN CMAKE SINCE THIS ADDS A NEW FILE TO THE SOURCE.

## Changes Proposed:
-  Move Chapter 3 spell of scarlet enclave to chapter 3 file
-  Establish Chapter 3 script of scarlet enclave

## Issues Addressed:

- Closes 

## Tests Performed:

- builds
- works

## How to Test the Changes:

1. nothing really to test. just moving the scripts for organizational purposes per source notes.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by pklloveyou** [comment]
Tested

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter3.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_q12779_an_end_to_all_things_SpellScript();
+    }
+};
+
+void AddSC_the_scarlet_enclave_c3()
+{
+    new spell_q12779_an_end_to_all_things();
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/ScarletEnclave/chapter3.cpp:52]: use RegisterSpellScript instead

**Comment by Nefertumm** [comment]
Hey! Move the spell script to the new system, please :D

File: src/server/scripts/EasternKingdoms/ScarletEnclave/chapter3.cpp (hunk 1 above)
Comment (acidmanifesto) [reply src/server/scripts/EasternKingdoms/ScarletEnclave/chapter3.cpp]: 420bf8a

**Comment by acidmanifesto** [comment]
> Hey! Move the spell script to the new system, please :D


420bf8a

**Review by Kitzunu — APPROVED** [review]

**Comment by acidmanifesto** [comment]
Thank you @Kitzunu and @Nefertumm for your review on the PR.

**Pull request merged**